### PR TITLE
va-file-input: show modal to confirm file delete

### PR DESCRIPTION
**Pull request by it-harrison**
## Chromatic

https://{{head.ref}}--65a6e2ed2314f7b8f98609d8.chromatic.com

---

## Description
This PR adds a modal to va-file-input that opens when a user clicks on the "delete" button after adding a file; the delete operation must be confirmed in the modal before being performed. 

Closes [2906](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2906)

## QA Checklist
- [ ] Component maintains 1:1 parity with design mocks
- [ ] Text is consistent with what's been provided in the mocks
- [ ] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ ] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [ ] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots

Screenshot 2024-06-14 at 2 27 58 PM

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts [hunk 1]
```diff
@@ -247,4 +247,106 @@ describe('va-file-input', () => {
...
+    
+    // get buttons again
+    const [__, deleteButton2] = await page.findAll('va-file-input >>> va-button-icon');
+      
+    // file not deleted because delete option still here
+    expect(deleteButton2).not.toBeNull();
+  });
+
+  // this test usually passes but it is too flaky to enable
+  it.skip('v3 opens a modal when delete button clicked and lets user remove the file', async () => {
```
Comment (Andrew565) [inline packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts:306]: Was this test also intentionally skipped?

File: packages/web-components/src/components/va-file-input/va-file-input.tsx [hunk 2]
```diff
@@ -154,12 +155,28 @@ export class VaFileInput {
...
     this.file = null;
     this.vaChange.emit({ files: [this.file] });
     this.uploadStatus = 'idle';
     this.internalError = null;
   };
 
+  private openModal = () => {
+    // set the status attribute here not in markup or it will have no effect
+    const modal = this.el.shadowRoot.querySelector('va-modal');
+    modal.setAttribute('status', 'warning');
```
Comment (Andrew565) [inline packages/web-components/src/components/va-file-input/va-file-input.tsx:168]: Does this warning status attribute need to be set manually here, or could it just be permanently set on the `va-modal` component itself?

Comment (it-harrison) [reply packages/web-components/src/components/va-file-input/va-file-input.tsx:168]: @Andrew565 I believe it must be set like this - if set on the component it does not have any effect (it doesn't get added to the markup on render) - we encountered something like this with va-button components inside other components, where we had to programmatically set the uswds prop. It's some kind of limitation with stencil and nested web-components.

File: packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts [hunk 3]
```diff
@@ -247,4 +247,106 @@ describe('va-file-input', () => {
 
     await axeCheck(page);
   });
+
+  // this test usually passes but it is too flaky to enable
+  it.skip('v3 opens a modal when delete button clicked and lets user remove or keep file', async () => {
```
[the remark this answers is not in the document]
Comment (it-harrison) [reply packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts:252]: @Andrew565 - yes, I could not get the test to run reliably - different errors seemingly at random but sometimes it passes - so I'm not sure if this feature can be tested (I noted there was another skipped test in this component); I left the test in as a reference

File: packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts (hunk 1 above)
Comment (it-harrison) [reply packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts:306]: @Andrew565 - yes, I could not get the test to run reliably - different errors seemingly at random but sometimes it passes - so I'm not sure if this feature can be tested (I noted there was another skipped test in this component); I left the test in as a reference

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**